include/hal/library/cryptlib/cryptlib_hkdf.h

```diff
@@ -45,6 +45,27 @@ extern bool libspdm_hkdf_sha256_extract(const uint8_t *key, size_t key_size,
 extern bool libspdm_hkdf_sha256_expand(const uint8_t *prk, size_t prk_size,
                                        const uint8_t *info, size_t info_size,
                                        uint8_t *out, size_t out_size);
+
+/**
+ * Derive key data using HMAC-SHA256 based KDF.
+ *
+ * @param[in]   key        Pointer to the user-supplied key.
+ * @param[in]   key_size   Key size in bytes.
+ * @param[in]   salt       Pointer to the salt value.
+ * @param[in]   salt_size  Salt size in bytes.
+ * @param[in]   info       Pointer to the application specific info.
+ * @param[in]   info_size  Info size in bytes.
+ * @param[out]  out        Pointer to buffer to receive hkdf value.
+ * @param[in]   out_size   Size of hkdf bytes to generate.
+ *
+ * @retval true   Hkdf generated successfully.
+ * @retval false  Hkdf generation failed.
+ **/
+extern bool libspdm_hkdf_sha256_extract_and_expand(const uint8_t *key, size_t key_size,
```

Comment:
Since `libspdm_hkdf_sha256_extract_and_expand` is only used for FIPS testing, and nothing else uses is, you can keep the function declaration in `os_stub/spdm_crypt_ext_lib/cryptlib_ext.h` and remove `libspdm_hkdf_sha256_extract_and_expand` from https://github.com/DMTF/libspdm/blob/9ad78c33235296c812e503e84ec7e5a89ba2e19d/library/spdm_crypt_lib/fips/libspdm_selftest_hkdf.c#L85-L88.

Reply:
Thanks. I have moved the `libspdm_hkdf_sha256_extract_and_expand `API to `os_stub`.